File: 0572-subtree-of-another-tree/0572-subtree-of-another-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean is = false;

    public boolean isSubtree(TreeNode root, TreeNode subRoot) {
        if (root != null && is == false)
        {
            //System.out.println(root.val + " " + subRoot.val);
            if (root.val == subRoot.val)
            {
                boolean m = parseSmall(root, subRoot);
                //System.out.println(m + " val");
                if (m == true)
                {
                    is = true;
                }
            }
            isSubtree(root.left, subRoot);
            isSubtree(root.right, subRoot);
        }
        return is;
    }

    // public void parseBig(TreeNode root, TreeNode subRoot)
    // {
    //     if (root == null)
    //     {
    //         return;
    //     }
    //     if (root.val == subRoot.val)
    //     {
    //         boolean m = parseSmall(root, subRoot);
    //         if (m == true)
    //         {
    //             //is = true
    //             return;
    //         }
    //     }
    // }

    public boolean parseSmall(TreeNode root, TreeNode subRoot)
    {
        if (root == null && subRoot == null)
        {
            return true;
        }
        if (root != null && subRoot == null)
        {
            return false;
        }
        if (root == null && subRoot != null)
        {
            return false;
        }
        boolean l = parseSmall(root.left, subRoot.left);
        if (l == false)
        {
            return false;
        }
        boolean r = parseSmall(root.right, subRoot.right);
        if (r == false)
        {
            return false;
        }
        return (root.val == subRoot.val);
    }
}

File: 0572-subtree-of-another-tree/0572-subtree-of-another-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
def isSameTree(root, subRoot):
        if root == None and subRoot == None:
            return True
        if (root == None and subRoot != None) or (root != None and subRoot == None) or root.val != subRoot.val:
            return False
        return isSameTree(root.left, subRoot.left) and isSameTree(root.right, subRoot.right)
class Solution(object):
    def isSubtree(self, root, subRoot):
        """
        :type root: Optional[TreeNode]
        :type subRoot: Optional[TreeNode]
        :rtype: bool
        """
        if root == None:
            return False
        if isSameTree(root, subRoot):
            return True
        return self.isSubtree(root.left, subRoot) or self.isSubtree(root.right, subRoot)